motionplan/ik/nloptInverseKinematics.go

```diff
@@ -126,10 +126,20 @@ func (ik *NloptIK) Solve(ctx context.Context,
 
 		if len(gradient) > 0 {
 			for i := range gradient {
+				flip := false
 				x[i] += ik.jump
+				if x[i] >= ik.upperBound[i] {
+					flip = true
+					x[i] -= 2 * ik.jump
+				}
+
 				inputs = referenceframe.FloatsToInputs(x)
 				eePos, err := ik.model.Transform(inputs)
-				x[i] -= ik.jump
+				if flip {
+					x[i] += ik.jump
```

Comment:
confused why we're mutating x[i] instead of copying it.  It seems like this transformation would not be necessary in that case